types/json/binary_functions.go

```diff
@@ -170,7 +170,7 @@ func (bj BinaryJSON) extractTo(buf []BinaryJSON, pathExpr PathExpression) []Bina
 	currentLeg, subPathExpr := pathExpr.popOneLeg()
 	if currentLeg.typ == pathLegIndex {
 		if bj.TypeCode != TypeCodeArray {
-			if currentLeg.arrayIndex <= 0 {
+			if currentLeg.arrayIndex <= 0 && currentLeg.arrayIndex != arrayIndexAsterisk {
```

Comment:
why change it?

Reply:
if I don‘t change this, query `select json_contains_path('{"a": 1, "b": 2, "c": {"d": 4}}', 'one', '$[*]')` will return 1 because `[*]` path will also extract non-array fields, in mysql `select json_contains_path('{"a": 1, "b": 2, "c": {"d": 4}}', 'one', '$[*]')` should return 0.